api/server/router/build/build_routes.go

```diff
@@ -173,8 +190,8 @@ func (br *buildRouter) postBuild(ctx context.Context, w http.ResponseWriter, r *
 	buildOptions.AuthConfigs = getAuthConfigs(r.Header)
 
 	if buildOptions.Squash && !br.daemon.HasExperimental() {
-		return apierrors.NewBadRequestError(
-			errors.New("squash is only supported with experimental mode"))
+		return validationError{
+			errors.New("squash is only supported with experimental mode")}
```

Comment:
nit: I'd either put this all, on the same line, or put the closing brace on its own line.